src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -126,42 +126,52 @@ public Delegate Compile(bool preferInterpretation)
                 return new Interpreter.LightCompiler().CompileTop(this).CreateDelegate();
             }
 #endif
-            return Compiler.LambdaCompiler.Compile(this);
+            return Compiler.LambdaCompiler.Compile(this, null);
 #else
             return new Interpreter.LightCompiler().CompileTop(this).CreateDelegate();
 #endif
         }
 
+        /// <summary>
+        /// Produces a delegate that represents the lambda expression.
+        /// </summary>
+        /// <param name="debugInfoGenerator">Debugging information generator used by the compiler to mark sequence points and annotate local variables.</param>
+        /// <returns>A delegate containing the compiled version of the lambda.</returns>
+        public Delegate Compile(DebugInfoGenerator debugInfoGenerator)
```

Comment:
>debugInfoGenerator [](start = 51, length = 18)

We should assert that this is not null.

Reply:
See [https://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/LambdaExpression.cs,138](https://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/LambdaExpression.cs,138).

---
In reply to: [296329629](https://github.com/dotnet/corefx/pull/38749#discussion_r296329629) [](ancestors = 296329629)